packages/cli/src/Validate.ts

```diff
@@ -62,7 +68,21 @@ ${bold('Examples')}
       return this.help()
     }
 
-    await loadEnvFile({ schemaPath: args['--schema'], printMessage: true })
+    const { config } = await loadConfigFromFile({ configFile: args['--config'] })
+
+    if (config?.env) {
+      if (config.env.kind === 'skip') {
+        console.debug(`Prisma config detected, skipping environment variable loading`)
+      } else if (config.env.kind === 'load') {
+        const env = await config.env.loadEnv()
+
+        // Note: Prisma currently relies on accessing `process.env` directly
+        process.env = { ...process.env, ...env }
+        console.debug(`Environment variables loaded via Prisma config`)
+      }
+    } else {
+      await loadEnvFile({ schemaPath: args['--schema'], printMessage: true })
+    }
```

Comment:
TODO: extract this as its own function + test it in a follow-up PR.
See [ORM-607](https://linear.app/prisma-company/issue/ORM-607/when-prismaconfigts-is-detected-disable-automatic-env-file-loading).